### PR TITLE
Most and least accurate

### PR DESCRIPTION
**Pull request by Rostammahabadi**
created method for most and least accurate teams for a specified season

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -7,6 +7,14 @@ def self.all
     @@all
   end
 
+  def find_by(id)
```
Comment (perryr16) [inline lib/game.rb:10]: useful tool for navigating between csv files

File: lib/game_team.rb [hunk 2]
```diff
@@ -37,6 +41,58 @@ def self.winningest_coach
...
+    self.find_by(season)
+  end
+
+  def self.get_goal_shots_by_game_team(game_teams)
+    #passes in an array of games
+    results = {}
+    #each game, iterating through and adding the team_id as the key and
+    #getting the shots and goals count
+    game_teams.each do |game_team|
+      results[game_team.team_id] ||= {"shots"=>0,"goals"=>0}
```
Comment (perryr16) [inline lib/game_team.rb:55]: does `||=` assign `{"shots"=>0,"goals"=>0}` if the array key is nil?

File: lib/game_team.rb [hunk 3]
```diff
@@ -37,6 +41,58 @@ def self.winningest_coach
...
+      results[game_team.team_id]["goals"] += game_team.goals
+    end
+    results
+  end
+
+  def self.least_accurate_team(season)
+   seasonal_hash = gets_team_shots_goals_count(season)
+   seasonal_hash.map do |key,value|
+     value["average"] = (value["shots"]/ value["goals"].to_f).round(2)
+   end
```
Comment (perryr16) [inline lib/game_team.rb:66]: cool way to mult/divide two hashes

File: test/game_team_test.rb [hunk 4]
```diff
@@ -100,4 +100,59 @@ def test_it_can_find_winninest_coach
...
   def test_it_can_find_worst_coach
   end
+
+  def test_find_by_returns_games
+    assert_kind_of Array, GameTeam.find_by(2012030221)
+  end
+
+  def test_game_team_shots_goals_count
+    Game.from_csv('./test/fixtures/games_20.csv')
+    # arr_games = Game.all[0..2]
```
Comment (perryr16) [inline test/game_team_test.rb:110]: look at those mocks and stubs!

File: lib/game_team.rb [hunk 5]
```diff
@@ -37,6 +41,58 @@ def self.winningest_coach
...
+  end
+
+  def self.gets_team_shots_goals_count(season)
+    #passes in desired season, grabs the *games* for the season
+    season_games = Game.grouped_by_season(season)
+    matches = []
+    #gets the game_teams associated with the season
+    season_games.each do |game|
+      matches.concat(GameTeam.find_by(game.game_id))
+    end
```
Comment (perryr16) [inline lib/game_team.rb:91]: I think this is a great use of data gathering between 2 csv files.

**Review by perryr16 — APPROVED** [review]
approved. It seems there are some branch conflicts.

File: lib/game_team.rb (hunk 2 above)
Comment (Rostammahabadi) [reply lib/game_team.rb:55]: Kind of. What it's doing is saying if the values don't exist for the key `results[game_team.team_id]` then assign the hash `{"shots"=>0,"goals"=>0}` as it's values creating a subhash or hash within a hash aka hashception

Comment (stellakunzang) [reply lib/game_team.rb:55]: memoization in the house!

File: test/game_team_test.rb [hunk 6]
```diff
@@ -100,4 +100,59 @@ def test_it_can_find_winninest_coach
...
+    game_team1.stubs(:goals).returns(3)
+    game_team2.stubs(:goals).returns(3)
+    game_team3.stubs(:goals).returns(2)
+    game_team4.stubs(:goals).returns(2)
+    game_team1.stubs(:shots).returns(6)
+    game_team2.stubs(:shots).returns(6)
+    game_team3.stubs(:shots).returns(4)
+    game_team4.stubs(:shots).returns(4)
+    passed_array = [game_team1,game_team2,game_team3,game_team4]
+
```
Comment (stellakunzang) [inline test/game_team_test.rb:141]: This is a great example of how to test with mocks and stubs!

File: lib/game.rb (hunk 1 above)
Comment (foleymichelle9) [reply lib/game.rb:10]: Oh nice! This will come in handy!

File: lib/game_team.rb (hunk 5 above)
Comment (foleymichelle9) [reply lib/game_team.rb:91]: Agreed!

File: test/game_team_test.rb (hunk 6 above)
Comment (foleymichelle9) [reply test/game_team_test.rb:141]: Agreed!

**Pull request merged**